Source/Core/Common/Arm64Emitter.h

```diff
@@ -920,6 +944,211 @@ class ARM64XEmitter
   bool TryORRI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm);
   bool TryEORI2R(ARM64Reg Rd, ARM64Reg Rn, u64 imm);
 
+  // For AND/TST/ORR/EOR etc
+  constexpr LogicalImm EncodeLogicalImm(u64 value, u32 width)
```

Comment:
This sounds like a good constructor of `LogicalImm` to me. Feel free to chose between explicit or implicit one.